parser.c

```diff
@@ -1015,3 +1015,31 @@ void validate_access_oTfd(int timeidx, int pos, int line, const char * function,
 		logg("             found in %s() (line %i) in %s", function, line, file);
 	}
 }
+
+void reresolveHostnames(void)
+{
+	int clientID;
+	for(clientID = 0; clientID < counters.clients_MAX; clientID++)
+	{
+		// Process this client only if it has at least one active query in the log
+		if(clients[clientID].count < 1)
+			continue;
+
+		// Get client hostname
+		char *hostname = resolveHostname(clients[clientID].ip);
+		if(strlen(hostname) > 0)
+		{
+			// Delete possibly already existing hostname pointer before storing new data
+			if(clients[clientID].name != NULL)
+			{
+				free(clients[clientID].name);
+				clients[clientID].name = NULL;
```

Comment:
Setting it to null isn't necessary as you set it to the new hostname immediately after this.

Reply:
It is good practice to set a pointer to `NULL` after `free`ing it and I'd like to keep it even though we may be setting it to a new value afterwards. I guess the compiler is going to optimize it away anyway in the compiled binaries.

> On a big, long-lived project, there are good reasons to set pointers to NULL
https://stackoverflow.com/a/1879678/2087442

Reply:
Although I think that post is more focused on freed variables that won't be reassigned, there are some valid arguments there.